addons/base_import/models/base_import.py

```diff
@@ -205,7 +203,12 @@ def _match_headers(self, rows, fields, options):
             for index, header in enumerate(headers)
         )
 
+    @api.v7
     def parse_preview(self, cr, uid, id, options, count=10, context=None):
+        return self.browse(cr, uid, id, context).parse_preview(options, count)
```

Comment:
I think the API split is not necessary here: `@api.multi` should work correctly as it will just `_normalize_ids` internally, which can take either an iterable of ids or a single "atomic" id (int, long, str or unicode). Same for `do`.